path_dynamic_roles.go

```diff
@@ -0,0 +1,317 @@
+package openldap
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/go-ldap/ldif"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/parseutil"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	secretCredsType = "creds"
+
+	dynamicRolePath = "role/"
+	dynamicCredPath = "cred/"
+)
+
+func (b *backend) pathDynamicRoles() []*framework.Path {
+	return []*framework.Path{
+		// POST/GET/DELETE role/:name
+		{
+			Pattern: path.Join(dynamicRolePath, framework.GenericNameRegex("name")),
+			Fields: map[string]*framework.FieldSchema{
+				"name": {
+					Type:        framework.TypeLowerCaseString,
+					Description: "Name of the role (lowercase)",
+					Required:    true,
+				},
+				"creation_ldif": {
+					Type:        framework.TypeString,
+					Description: "LDIF string used to create new entities within OpenLDAP. This LDIF can be templated.",
+					Required:    true,
+				},
+				"deletion_ldif": {
+					Type:        framework.TypeString,
+					Description: "LDIF string used to delete entities created within OpenLDAP. This LDIF can be templated.",
+					Required:    true,
+				},
+				"rollback_ldif": {
+					Type:        framework.TypeString,
+					Description: "LDIF string used to rollback changes in the event of a failure to create credentials. This LDIF can be templated.",
+				},
+				"username_template": {
+					Type:        framework.TypeString,
+					Description: "The template used to create a username",
+				},
+				"default_ttl": {
+					Type:        framework.TypeDurationSecond,
+					Description: "Default TTL for dynamic credentials",
+				},
+				"max_ttl": {
+					Type:        framework.TypeDurationSecond,
+					Description: "Max TTL a dynamic credential can be extended to",
+				},
+			},
+			ExistenceCheck: b.pathDynamicRoleExistenceCheck,
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: b.pathDynamicRoleCreateUpdate,
+				},
+				logical.CreateOperation: &framework.PathOperation{
+					Callback: b.pathDynamicRoleCreateUpdate,
+				},
+				logical.ReadOperation: &framework.PathOperation{
+					Callback: b.pathDynamicRoleRead,
+				},
+				logical.DeleteOperation: &framework.PathOperation{
+					Callback:                    b.pathDynamicRoleDelete,
+					ForwardPerformanceStandby:   true,
```

Comment:
Is there a reason `ForwardPerformanceStandby` and `ForwardPerformanceSecondary` are only on the `DeleteOperation`?

Reply:
Not specifically. I started by copy-pasting the static role definitions which only have these set on `DeleteOperation`. My knowledge of these fields is pretty limited - should they be set on all of the write operations?

Reply:
I don't believe this is needed at all here. The general rule for using `ForwardPerformanceStandby` and `ForwardPerformanceSecondary` is if the method interacts and modifies an external system, e.g. rotating credentials for a root account or something. Docs: 

- https://github.com/hashicorp/vault/blob/22008eefb69cbd6ad88ec620f83b4f8aaa59b3bc/sdk/framework/path.go#L156-L178

Reply:
Gotcha - that makes sense. Thanks for the info! I've updated the PR to remove these.